homework/zadanie4_five_philosophers.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <thread>
+#include <vector>
+#include <chrono>
+#include <mutex>
+#include <string>
+#include <mutex>
+
+using namespace std::chrono_literals;
+
+std::mutex mtx;
+
+class CTable{
+
+public:
+    std::vector<bool> forks_pool;
+
+    CTable(int number_of_forks){
+        for (size_t i=0; i< number_of_forks; i++){
+            forks_pool.push_back(false);
+        }
+    }
+};
+
+class CPhilosoph{
+
+public:
+    std::string name_;
+    std::string output;
+    int number_;
+    CTable table_;
```

Comment:
Każdy filozof ma swoją kopię CTable. Referencji brakuje.

Reply:
Tak, zauwazylem to po ponownym przejrzeniu kodu, idiotyczne to bylo, zgadzam sie, teraz mamy jeden stol, kazdy filo ma adres tego stolu.